app/controllers/gallery/albums_controller.rb

```diff
@@ -0,0 +1,12 @@
+# encoding:UTF-8
+class Gallery::AlbumsController < ApplicationController
+  load_permissions_and_authorize_resource
+
+  def show
+    if params[:start].present?
+      @start = params[:start]
+    elsif params[:image].present?
+      @start = @album.images.find_index(Image.find(params[:image]))
+    end
```

Comment:
You are not handling the case where ```!(params[:start].present? || params[:image].present?)```. Since params come directly from the user it's not prudent to make any assumptions about them.